realsense_camera/src/debug_tool.cpp

```diff
@@ -0,0 +1,124 @@
+ specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+#include "ros/ros.h"
+#include "std_msgs/String.h"
+#include <librealsense/rs.hpp>
+#include <librealsense/rs.h>
+#include <string>
+#include <sstream>
+#include <ros/package.h>
+#include <sys/utsname.h>
+#include <cstdlib>
+#include <stdlib.h>
+#include <array>
+
+rs_context *rs_context_ = NULL;
+rs_error *rs_error_ = NULL;
+
+std::string getSystemCallOutput(const char* cmd)
+{
+  std::array<char, 128> buffer;
+  std::string result;
+  std::shared_ptr<FILE> pipe(popen(cmd, "r"), pclose);
+  if (!pipe) throw std::runtime_error("popen() failed!");
+  while (!feof(pipe.get()))
+  {
+    if (fgets(buffer.data(), 128, pipe.get()) != NULL)
+    {
+      result += buffer.data();
+    }
+  }
+  std::string::size_type pos = result.find_last_not_of("\n \t");
+  if (pos != std::string::npos)
+  {
+    result = result.substr(0, pos + 1);
+  }
+  return result;
+}
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "debug_tool");
+  ros::NodeHandle n;
+  std::string message = "\n----------CUT-PASTE-THIS----------\n";
+  message = message + "|Version | Your Configuration | \n|:----------|:----------| \n";
+
+  /* Acquire OS */
+  message = message + "|Operating System | ";
+  std::string operating_system = getSystemCallOutput("lsb_release -ds");
+  message = message + operating_system + "|\n";
+
+  /* Acquire Kernel Release */
+  struct utsname *buf = new utsname();
+  uname(buf);
+  message = message + "|Kernel | ";
+  std::string kernel = buf->release;
+  message = message + kernel + "|\n";
+
+  /* Acquire ROS Version */
+  message = message + "|ROS | ";
+  std::string ros_version = std::getenv("ROS_DISTRO");
+  message = message + ros_version + "| \n";
+
+  /* Acquire ROS RealSense Version */
+  message = message + "|ROS RealSense | ";
+  std::string realsense_version = getSystemCallOutput("rosversion realsense_camera");
+  message = message + realsense_version + "|\n";
+
+  /* Acquire librealsense version */
+  message = message + "|librealsense | ";
+  std::string librealsense_version = getSystemCallOutput("rosversion librealsense");
```

Comment:
We should add try/catch around each call to `getSystemCallOutput` in case the command fails; the catch can provide a "fail" messages in the output.